flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/TestingLeaderElectionEventHandler.java

```diff
@@ -74,6 +75,15 @@ private void ifRunning(Runnable action) {
         }
     }
 
+    @Override
+    public CompletableFuture<Void> onGrantLeadershipAsync(UUID newLeaderSessionId) {
```

Comment:
How about marking `TestingLeaderElectionEventHandler` and `TestingLeaderContender` as `@Deprecated` first to prevent other developers from using this class for testing and reduce the workload of ultimately removing these two classes. 🤔

Reply:
I'm also not happy about having all the different testing implementations. The two you mentioned have actually a purpose, though. They implement basic functionality that would be annoying to implement for every test case using the new `TestingGeneric*` implementations. Therefore, I would suggest leaving them as is.

Reply:
After looking at the relevant code again, It make sense to leave them as is.